nixos/modules/services/audio/mpdscribble.nix

```diff
@@ -0,0 +1,231 @@
+          "2" logs informational messages (e.g.  "new song");
+          "3" prints  a lot  of debugging messages.
+        '';
+      };
+
+      proxy = mkOption {
+        default = "";
+        example = ''http://your.proxy:8080'';
+        type = type.string;
+        description = ''
+          HTTP Proxy URL.
+        '';
+      };
+
+      lastfm = {
+
+        enable = mkOption {
+          default = false;
+          description = ''
+            Enable Lastfm scrobbling.
+          '';
+        };
+
+        url = mkOption {
+          default = ''http://post.audioscrobbler.com/'';
+          example = ''http://post.audioscrobbler.com/'';
+          type = type.string;
+          description = ''
+            The handshake URL of the scrobbler.
+          '';
+        };
+
+        username = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Last.fm User
+          '';
+        };
+
+        password = mkOption {
+          default = "";
+          example = "";
+          type = type.;
+          description = ''
+            Last.fm Password
+          '';
+        };
+
+        journal = mkOption {
+          default = "";
+          example = /home/user/.mpdscribble.lastfm.cache;
+          type = type.path;
+          description = ''
+            (Optional) file path for caching scrobbles when no internet connection
+            is available.
+          '';
+        };
+
+      };
+
+      librefm = {
+
+        enable = mkOption {
+          default = false;
+          description = ''
+            Enable Librefm scrobbling.
+          '';
+        };
+
+        url = mkOption {
+          default = ''http://turtle.libre.fm/'';
+          example = ''http://turtle.libre.fm/'';
+          type = type.string;
+          description = ''
+            The handshake URL of the scrobbler.
+          '';
+        };
+
+        username = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Librefm user.
+          '';
+        };
+
+        password = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Librefm Password
+          '';
+        };
+
+        journal = mkOption {
+          default = "";
+          example = /home/user/.mpdscribble.librefm.cache;
+          type = type.path;
+          description = ''
+            (Optional) file path for caching scrobbles when no internet connection
+            is available.
+          '';
+        };
+
+      };
+
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.mpdscribble = {
+      after         = [ "mpd.service" ];
+      description   = "MPD scrobble client for lastfm/librefm";
+      wants         = [ "mpd.service" ];
+      path          = [ pkgs.mpdscribble ];
+      script        = "exec mpdscribble";
```

Comment:
I believe you ought to pass --no-daemon (?) or set serviceConfig.Type = "forking"